financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/claims/FSClaimProvider.java

```diff
@@ -82,23 +82,39 @@ public Map<String, Object> getAdditionalClaims(OAuthTokenReqMessageContext token
                                                    OAuth2AccessTokenRespDTO tokenRespDTO)
             throws IdentityOAuth2Exception {
 
+        Map<String, Object> additionalClaims;
         if (ServiceExtensionUtils.isInvokeExternalService(ServiceExtensionTypeEnum
                 .PRE_ID_TOKEN_GENERATION)) {
             // Perform FS customized behaviour with service extension
             try {
-                return additionalIdTokenClaimsTokenResponseWithServiceExtension(
+                additionalClaims = additionalIdTokenClaimsTokenResponseWithServiceExtension(
                         tokenReqMessageContext, tokenRespDTO);
             } catch (FinancialServicesException e) {
                 log.error("Error while invoking external service extension", e);
                 throw new IdentityOAuth2Exception("Error while invoking external service extension");
             }
         } else if (getClaimProvider() != null) {
             // Perform FS customized behaviour
-            return getClaimProvider().getAdditionalClaims(tokenReqMessageContext, tokenRespDTO);
+            additionalClaims = getClaimProvider().getAdditionalClaims(tokenReqMessageContext, tokenRespDTO);
         } else {
             // Perform FS default behaviour
-            return defaultAdditionalIdTokenClaimsTokenResponse(tokenReqMessageContext, tokenRespDTO);
+            additionalClaims = defaultAdditionalIdTokenClaimsTokenResponse(tokenReqMessageContext, tokenRespDTO);
         }
+
+        tokenRespDTO.setAuthorizedScopes(updateScopeInTokenResponse(tokenRespDTO.getAuthorizedScopes()));
+        return additionalClaims;
+    }
+
+    private String updateScopeInTokenResponse(String scopes) {
```

Comment:
Add a comment on what is updated

Reply:
Fixed with 533d2b6e24e4c38d24460cd3042132849e579b08